src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1703,7 +1707,7 @@ abstract class RefChecks extends Transform {
               assert(sym != NoSymbol, "transformCaseApply: name = " + name.debugString + " tree = " + tree + " / " + tree.getClass) //debug
               enterReference(tree.pos, sym)
             }
-            tree
+            eliminatedRightAssocValDefs.getOrElse(tree.symbol, tree)
```

Comment:
OK, I see. It's owned by the `ValDef` that gets eliminated, so I have to change it anyway, even within the same block.